### PR TITLE
Allow overriding Electron app paths

### PR DESCRIPTION
**Pull request by inversion**
This is useful for example to change the userData directory so you don't reuse cookies and such from other instances.

### CODE REVIEW

**Comment by alfonsomga** [comment]
Merge this asap please...

**Comment by rosshinkley** [comment]
+1

**Comment by rhysd** [comment]
This is greate for testing Electron apps!

File: lib/nightmare.js [hunk 1]
```diff
@@ -49,7 +49,14 @@ function Nightmare(options) {
   options = options || {};
   var self = this;
 
-  this.proc = proc.spawn(electron_path, [runner], {
+  var electronArgs = [runner];
+
+  if(options.paths) {
+
+  	electronArgs.push( JSON.stringify(options.paths) );
```
Comment (matthewmueller) [inline lib/nightmare.js]: strange spacing. please make this code consistent with the rest of the code

File: lib/runner.js [hunk 2]
```diff
@@ -11,6 +11,17 @@ var fs = require('fs');
 var defaults = require('defaults');
 var join = require('path').join;
 
+
+if( process.argv.length > 2 ) {
```
Comment (matthewmueller) [inline lib/runner.js]: spacing

File: lib/runner.js [hunk 3]
```diff
@@ -11,6 +11,17 @@ var fs = require('fs');
...
+
+  var paths = JSON.parse(process.argv[2]);
+
+  for(var i in paths) {
+
+    app.setPath( i, paths[i] );
```
Comment (matthewmueller) [inline lib/runner.js]: spacing

**Comment by matthewmueller** [comment]
please fix up the spacing and this should be good to go!

**Comment by inversion** [comment]
@matthewmueller are there any autoformatter settings for the project? If we could standardise them it would help.

**Comment by inversion** [comment]
Spacing updated @matthewmueller

**Comment by matthewmueller** [comment]
@inversion yep, good call. planning on adding standard.

**Comment by matthewmueller** [comment]
now on master, thanks @inversion !

**Pull request merged**